### ISSUE TITLE
Using LsqFit with several x variables?

### ISSUE THREAD

**Issue by skanskan** [JuliaNLSolvers/LsqFit.jl#95]
Hello 

I'm trying to find a Julia pacakge faster than GLM.jl

How would you use it to fit a simple linear regression with several variables?

```
N=1000
x1 = repeat(1:N, outer=N)
x2 = repeat(1:N, inner=N)
x3 = sqrt.(repeat(1:N^2))
x1x2 = x1 .* x2
gg = repeat(1:5,inner=trunc(Int,N^2/5))
y = 1 .- 2*x1 + 3*x2 + 0.5*x1x2 + rand(N^2) + x3.*rand(N^2)
data = DataFrame(x1=x1, x2=x2, x3=x3, x1x2=x1x2, y=y,gg=gg)
categorical!(data, :gg)

```
This is the way GLM does it:
`fit(LinearModel, @formula(y ~ x1+x2+x3+x1x2), data)`

Or with MixedModels I can even use random effects (repeated measures)
`fit(LinearMixedModel, @formula(y ~x1+x2+x3+x1x2 + (1|gg)), data)`

How would you do it with LsqFit?

Basically I don't know the syntax to specify the model.

**Comment by spockoyno** [comment]
It should work.  Just put your variables in a  3 x N array. See the signature of `curve_fit(model::Function, xpts::AbstractArray, ydata::AbstractArray, p0; kwargs...)` in the source code.

**Comment by pkofod** [comment]
We should probably have an explicit example here.

**Comment by pkofod** [comment]
see the README.md now
